src/puppeteer_pool.js

```diff
@@ -388,44 +399,70 @@ class PuppeteerPool {
         });
     }
 
+    /**
+     * Updates the instance metadata when a new page is opened.
+     *
+     * @param {PuppeteerInstance} instance
+     * @ignore
+     */
+    _updateInstance(instance) {
```

Comment:
The function name could be more descriptive, e.g. `_onNewPageOpened()`

Reply:
or `_incrementPageCount()`